segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/noop/NoOpStorage.java

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.noop;
+
+import com.google.common.base.Preconditions;
+import com.google.common.util.concurrent.Uninterruptibles;
+import io.pravega.segmentstore.contracts.SegmentProperties;
+import io.pravega.segmentstore.contracts.StreamSegmentException;
+import io.pravega.segmentstore.storage.SegmentHandle;
+import io.pravega.segmentstore.storage.SyncStorage;
+import java.io.InputStream;
+import java.util.concurrent.TimeUnit;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static io.pravega.shared.NameUtils.INTERNAL_NAME_PREFIX;
```

Comment:
We should use `INTERNAL_SCOPE_NAME`.

While you're at it:
- Also use this in your unit test (I left a comment way below about it)
- Please modify `StreamSegmentNameUtils.java#64` to reference `INTERNAL_SCOPENAME` as a prefix instead of `_system`.

Reply:
While looking at this class, I realized that there may be more metadata tables out there that need to be included in the pass-through. If you look at `StreamSegmentNameUtils.getQualifiedTableName`, which is used by the Controller to generate names of its metadata tables, it stores this in a non-system segment which contains `_tables` (StreamSegmentNameUtils.TABLES) as the second token (position 1). I think we should modify `NoOpStorage` to allow this as pass-through as well (just check it if contains this substring. If we don't, we risk the Controller not being able to read its own Scope/Stream metadata which could quickly lead to weird behavior.

Reply:
Seems only checking the leading "_" just works for this scenario too.
As I explained above, in test codes we used too many short cut segment name instead of fully qualified names. But so far the bottom line ("_" is for internal segments) is protected.

Reply:
Added table segment as pass-through in NoOpStorage.

As for the INTERNAL_SCOPENAME, as I explained above, there are many test codes not using fully qualified segment names. If we think it's necessary now to enforce it, it worth a separate effort to convert all the disqualified segment into qualified in all test code.